sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java

```diff
@@ -85,15 +85,53 @@
 
   /** Information accessible to all methods in this {@code DoFn}. */
   public abstract class Context {
-
     /**
-     * Returns the {@code PipelineOptions} specified with the
-     * {@link org.apache.beam.sdk.runners.PipelineRunner}
-     * invoking this {@code DoFn}.  The {@code PipelineOptions} will
-     * be the default running via {@link DoFnTester}.
+     * Returns the {@code PipelineOptions} specified with the {@link
+     * org.apache.beam.sdk.runners.PipelineRunner} invoking this {@code DoFn}. The {@code
+     * PipelineOptions} will be the default running via {@link DoFnTester}.
      */
     public abstract PipelineOptions getPipelineOptions();
+  }
+
+  /**
+   * Information accessible while within the {@link StartBundle} method.
+   */
+  public abstract class StartBundleContext extends Context {
```

Comment:
Yes, let's forbid `Context` parameters and deprecate `Context` and `getPIpelineOptions`. We should provide `PipelineOptions` as its own parameter instead. Possibly before FSR, but let's leave open the ability to make the switch after FSR.